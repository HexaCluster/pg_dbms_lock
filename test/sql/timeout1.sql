-- Create an exlusive lock in the first session
DO $$
DECLARE
    v_result     integer;
    dbms_lock_x_mode integer := 6;
BEGIN

    v_result := dbms_lock.request(123, dbms_lock_x_mode);

    IF v_result <> 0 THEN
        RAISE NOTICE '%', (
           case 
              when v_result=1 then 'Timeout'
              when v_result=2 then 'Deadlock'
              when v_result=3 then 'Parameter Error'
              when v_result=4 then 'Already owned'
              when v_result=5 then 'Illegal Lock Handle'
            end);
    END IF;

END;
$$;


SELECT objid, mode FROM pg_locks WHERE objid IS NOT NULL;

SELECT pg_sleep(3);

-- now release the lock
DO $$
DECLARE
    v_result     integer;
BEGIN

    v_result := dbms_lock.release(123);

    IF v_result <> 0 THEN
        RAISE NOTICE '%', (
           case 
              when v_result=1 then 'Timeout'
              when v_result=2 then 'Deadlock'
              when v_result=3 then 'Parameter Error'
              when v_result=4 then 'Already owned'
              when v_result=5 then 'Illegal Lock Handle'
            end);
    END IF;

END;
$$;

SELECT objid, mode FROM pg_locks WHERE objid IS NOT NULL;

