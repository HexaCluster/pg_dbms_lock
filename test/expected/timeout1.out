DO
 objid |     mode      
-------+---------------
   123 | ExclusiveLock
(1 row)

 pg_sleep 
----------
 
(1 row)

DO
 objid | mode 
-------+------
(0 rows)

